source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -772,6 +772,23 @@ InjaTransformer::InjaTransformer(const TransformationTemplate &transformation,
     merged_extractors_to_body_ = true;
     break;
   }
+  case TransformationTemplate::kMergeJsonKeys: {
+    if (transformation.parse_body_behavior() == TransformationTemplate::DontParse) {
+      throw EnvoyException("MergeJsonKeys requires parsing the body");
+    } else if (transformation.advanced_templates()) {
+      throw EnvoyException("MergeJsonKeys is not supported with advanced templates");
+    } 
+    try {
+      for (const auto &named_extractor : transformation.merge_json_keys().json_keys()) {
+        merge_templates_.emplace_back(std::make_tuple(named_extractor.first, named_extractor.second.override_empty(), instance_->parse(named_extractor.second.tmpl().text())));
+      }
+
+    } catch (const std::exception &e) {
+      throw EnvoyException(
+          fmt::format("Failed to parse merge_body_key template {}", e.what()));
```

Comment:
what's merge_body_key? did you want to interpolate the key?

Reply:
Going to nest the exception to have access to the key